pylua/test/test_timer.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+
+import pylua.timer
+import logging
+import time
+import threading
+
+logger = logging.getLogger(__name__)
+
+class TestPyTimer:
+	__test_timer = None
+	__start_timer = None
+	__finish_timer = None
+	__interval = None
+	__cond_var = None
+
+	def timer_task(self):
+		self.__cond_var.acquire()
+		self.__finish_timer = float(time.time())
+		self.__cond_var.notify()
+		self.__cond_var.release()
+		pass
+
+	def __init__(self):
+		self.__interval = 1
+		self.__test_timer = timer.PyTimer(self.timer_task, self.__interval)
+		self.__cond_var = threading.Condition()
+		pass
+
+	def run_cond_var(self, wait_time):
+		self.__cond_var.acquire()
+		self.__cond_var.wait(wait_time)
+		self.__cond_var.release()
+		pass
+
+	def test_run_timer(self, interval = None):
+		logger.debug('test_run_timer_for_one_sec')
+		self.__finish_timer = None
+		self.__start_timer = float(time.time())
+		if (interval is None):
+			wait_time = self.__interval + 1
+			self.__test_timer.start()
+		else:
+			wait_time = interval + 1
+			self.__interval = interval
+			self.__test_timer.start(interval)
+		self.run_cond_var(wait_time)		    
+		assert isinstance(self.__finish_timer, float), "Timer doesn't work"
+		timer_infelicity = abs(self.__finish_timer - self.__start_timer - self.__interval)
+		logger.debug('timer_infelicity = ')
+		assert timer_infelicity < 0.05, "Timer doesn't work correctly"
+
+	def test_start_timer_with_parameter(self):
+		logger.debug('test_start_timer_with_parameter')
+		self.test_run_timer(2)
+		pass
+	
+	def test_start_timer_periodicaly_with_parameter(self):
+		logger.debug('test_start_timer_periodicaly_with_parameter')
```

Comment:
@malirod I checked using of timer from lua, and I saw that timer is used as single shot. I add functionality to PyTimer that allows to run timer several times from one object.